cmd/slinky/config.go

```diff
@@ -0,0 +1,274 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/skip-mev/slinky/cmd/constants"
+	"github.com/skip-mev/slinky/oracle/config"
+	mmtypes "github.com/skip-mev/slinky/service/clients/marketmap/types"
+)
+
+const (
+	// DefaultUpdateInterval is the default value for how frequently slinky updates aggregate price responses.
+	DefaultUpdateInterval = 250000000
+	// DefaultMaxPriceAge is the default value for the oldest price considered in an aggregate price response by slinky.
+	DefaultMaxPriceAge = 120000000000
+	// DefaultPrometheusServerAddress is the default value for the prometheus server address in slinky.
+	DefaultPrometheusServerAddress = "0.0.0.0:8002"
+	// DefaultMetricsEnabled is the default value for enabling prometheus metrics in slinky.
+	DefaultMetricsEnabled = true
+	// DefaultHost is the default for the slinky oracle server host.
+	DefaultHost = "0.0.0.0"
+	// DefaultPort is the default for the slinky oracle server port.
+	DefaultPort = "8080"
+	// jsonFieldDelimiter is the delimiter used to separate fields in the JSON output.
+	jsonFieldDelimiter = "."
+)
+
+// DefaultOracleConfig returns the default configuration for the slinky oracle.
+func DefaultOracleConfig() OracleConfig {
+	cfg := OracleConfig{
+		UpdateInterval: DefaultUpdateInterval,
+		MaxPriceAge:    DefaultMaxPriceAge,
+		Metrics: config.MetricsConfig{
+			PrometheusServerAddress: DefaultPrometheusServerAddress,
+			Enabled:                 DefaultMetricsEnabled,
+		},
+		Providers: make(map[string]config.ProviderConfig),
+		Host:      DefaultHost,
+		Port:      DefaultPort,
+	}
+
+	for _, provider := range append(constants.Providers, constants.MarketMapProviders...) {
+		cfg.Providers[provider.Name] = provider
+	}
+
+	return cfg
+}
+
+type OracleConfig struct {
+	// UpdateInterval is the interval at which the oracle will fetch prices from providers.
+	UpdateInterval time.Duration `json:"updateInterval"`
+
+	// MaxPriceAge is the maximum age of a price that the oracle will consider valid. If a
+	// price is older than this, the oracle will not consider it valid and will not return it in /prices
+	// requests.
+	MaxPriceAge time.Duration `json:"maxPriceAge"`
+
+	// Providers is the map of provider names to providers that the oracle will fetch prices from.
+	Providers map[string]config.ProviderConfig `json:"providers"`
+
+	// Production specifies whether the oracle is running in production mode. This is used to
+	// determine whether the oracle should be run in debug mode or not.
+	//
+	// Deprecated: This field is no longer used.
+	Production bool `json:"production"`
```

Comment:
This is effectively a new config which is incompatible with the existing config. We should just remove this in this PR.